llvm_mode/afl-llvm-pass.so.cc

```diff
@@ -87,88 +94,75 @@ bool AFLCoverage::runOnModule(Module &M) {
 
   } else be_quiet = 1;
 
-  /* Decide instrumentation ratio */
+  /* Get globals for the SHM region. */
 
-  char* inst_ratio_str = getenv("AFL_INST_RATIO");
-  unsigned int inst_ratio = 100;
-
-  if (inst_ratio_str) {
-
-    if (sscanf(inst_ratio_str, "%u", &inst_ratio) != 1 || !inst_ratio ||
-        inst_ratio > 100)
-      FATAL("Bad value of AFL_INST_RATIO (must be between 1 and 100)");
-
-  }
-
-  /* Get globals for the SHM region and the previous location. Note that
-     __afl_prev_loc is thread-local. */
-
-  GlobalVariable *AFLMapPtr =
-      new GlobalVariable(M, PointerType::get(Int8Ty, 0), false,
-                         GlobalValue::ExternalLinkage, 0, "__afl_area_ptr");
-
-  GlobalVariable *AFLPrevLoc = new GlobalVariable(
-      M, Int32Ty, false, GlobalValue::ExternalLinkage, 0, "__afl_prev_loc",
-      0, GlobalVariable::GeneralDynamicTLSModel, 0, false);
+  GlobalVariable *BBCoverageMapPtr =
+      new GlobalVariable(M, PointerType::get(Int64Ty, 0), false,
+                         GlobalValue::ExternalLinkage, 0, "__bb_coverage_ptr");
 
   /* Instrument all the things! */
 
-  int inst_blocks = 0;
+  u64 block_counter = 0;
+  std::ofstream location_file;
+  bool write_to_file = false;
```

Comment:
we don't really need this variable. We should be able to call location_file.is_open()?